### PR TITLE
added GridView and interactive option

### PR DESCRIPTION
**Pull request by AmitSilber**

### CODE REVIEW

File: app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java [hunk 1]
```diff
@@ -34,11 +34,28 @@ protected static String buildUrl(String[] ingridients) {
...
+
+        gridview.setOnItemClickListener(new OnItemClickListener() {
+            public void onItemClick(AdapterView<?> parent, View v,
+                                    int position, long id) {
+                ImageAdapter adapter = (ImageAdapter) gridview.getAdapter();
+                adapter.setIsPressed(position);
+                if (adapter.getIsPressed(position)) {
+                    ImageView imageView = (ImageView) v;
+                    imageView.setImageAlpha(100);
+                }else{
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java]: add spaces (`} else {`)

File: app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java [hunk 2]
```diff
@@ -34,11 +34,28 @@ protected static String buildUrl(String[] ingridients) {
...
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_choose_ingredients);
-        Toolbar toolbar = findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
+        setContentView(R.layout.content_choose_ingredients);
+
+        final GridView gridview = (GridView) findViewById(R.id.gridview);
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java:62]: can be extracted to outer method "setupGridView",
will make `onCreate` cleaner.

File: app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java [hunk 3]
```diff
@@ -34,11 +34,28 @@ protected static String buildUrl(String[] ingridients) {
...
+                ImageAdapter adapter = (ImageAdapter) gridview.getAdapter();
+                adapter.setIsPressed(position);
+                if (adapter.getIsPressed(position)) {
+                    ImageView imageView = (ImageView) v;
+                    imageView.setImageAlpha(100);
+                }else{
+                    ImageView imageView = (ImageView) v;
+                    imageView.setImageAlpha(255);
+                }
+
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ChooseIngredients.java:77]: remove emptyline

File: app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ImageView;
+
+import java.util.Arrays;
+
+public class ImageAdapter extends BaseAdapter {
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java:14]: add class documentation

File: app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        return isPressed[id];
+    }
+    // create a new ImageView for each item referenced by the Adapter
+
+    public View getView(int position, View convertView, ViewGroup parent) {
+        ImageView imageView;
+        if (convertView == null) {
+            // if it's not recycled, initialize some attributes
+            imageView = new ImageView(mContext);
+            //imageView.setLayoutParams(new ViewGroup.LayoutParams(100, 100));
```
Comment (mfocshaner) [inline app/src/main/java/com/huji/foodtricks/foodtricks/ImageAdapter.java]: remove

File: app/src/main/res/layout/content_choose_ingredients.xml [hunk 6]
```diff
@@ -8,7 +8,7 @@
     tools:context=".ChooseIngredients"
     tools:showIn="@layout/activity_choose_ingredients">
 
-    <TextView
+    <!--TextView
```
Comment (mfocshaner) [inline app/src/main/res/layout/content_choose_ingredients.xml:11]: just delete this

**Review by mfocshaner — CHANGES_REQUESTED** [review]
please fix the build.gradle changes, remove them from the PR

File: app/src/main/res/layout/content_choose_ingredients.xml (hunk 6 above)
Comment (AmitSilber) [reply app/src/main/res/layout/content_choose_ingredients.xml:11]: we'll need this for the "feed me" button

**Pull request merged**